thingsboard_gateway/connectors/ftp/path.py

```diff
@@ -0,0 +1,181 @@
+#     Copyright 2021. ThingsBoard
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+
+import os
+from regex import compile
+
+from thingsboard_gateway.connectors.ftp.file import File
+
+COMPATIBLE_FILE_EXTENSIONS = ('json', 'txt', 'csv')
+
+
+class Path:
+    def __init__(self, path: str, delimiter: str, telemetry: list, device_name: str, attributes: list,
+                 txt_file_data_view: str, poll_period=60, with_sorting_files=True, device_type='Device', max_size=5,
+                 read_mode='FULL'):
+        self._path = path
+        self._with_sorting_files = with_sorting_files
+        self._poll_period = poll_period
+        self._files: [File] = []
+        self._delimiter = delimiter
+        self._last_polled_time = 0
+        self._telemetry = telemetry
+        self._attributes = attributes
+        self._device_name = device_name
+        self._device_type = device_type
+        self._txt_file_data_view = txt_file_data_view
+        self.__read_mode = File.ReadMode[read_mode]
+        self.__max_size = max_size
+
+    @staticmethod
+    def __is_file(ftp, filename):
+        current = ftp.pwd()
+        try:
+            ftp.cwd(filename)
+        except Exception:
+            ftp.cwd(current)
+            return True
+        ftp.cwd(current)
+        return False
+
+    @staticmethod
+    def __folder_exist(ftp, folder_name):
```

Comment:
It looks the same like __is_file method, but reversed, could we create a type for object and return FOLDER or FILE?